pkg/resource/plugin/analyzer.go

```diff
@@ -31,13 +32,17 @@ type Analyzer interface {
 	// Name fetches an analyzer's qualified name.
 	Name() tokens.QName
 	// Analyze analyzes a single resource object, and returns any errors that it finds.
-	Analyze(t tokens.Type, props resource.PropertyMap) ([]AnalyzeFailure, error)
+	Analyze(t tokens.Type, props resource.PropertyMap) ([]AnalyzeDiagnostic, error)
```

Comment:
I'm not super enthusiastic about the little type conversion dance we do here. We did it to stay consistent with how we're handling other events, converting `*pulumirpc.AnalyzeDiagnostic` -> `AnalyzeDiagnostic` -> `AnalyzePayload`. This is way more complex than it seems like it needs to be.